File: hackerrank/easy/0038-cut-the-sticks/python/solution.py
## solution.py

from typing import List

def cut_the_sticks(arr: List[int]) -> List[int]:
    if not arr:
        return []
    a = sorted(arr)
    res = []
    i = 0
    n = len(a)
    while i < n:
        res.append(n - i)
        cur = a[i]
        while i < n and a[i] == cur:
            i += 1
    return res


# --- demo runs ---
if __name__ == "__main__":
    print(cut_the_sticks([5,4,4,2,2,8]))        # [6, 4, 2, 1]
    print(cut_the_sticks([1,2,3,4,3,3,2,1]))    # [8, 6, 4, 1]

File: hackerrank/easy/0038-cut-the-sticks/js/solution.mjs
// solution.mjs

export function cutTheSticks(arr) {
  if (!arr || arr.length === 0) return [];
  const a = [...arr].sort((x, y) => x - y);
  const res = [];
  let i = 0;
  while (i < a.length) {
    res.push(a.length - i);
    const cur = a[i];
    while (i < a.length && a[i] === cur) i++;
  }
  return res;
}

// --- demo runs ---
console.log(cutTheSticks([5,4,4,2,2,8]));       // [6, 4, 2, 1]
console.log(cutTheSticks([1,2,3,4,3,3,2,1]));   // [8, 6, 4, 1]